File: maxtemperature-monoid-folding/src/MaxTemperatureMonoidFolding/Series.cs
namespace MaxTemperatureMonoidFolding;

public class Series<T> {
    private T[] Values { get; set; }

    public Series(T[] values) {
        Values = values;
    }

    public T Reduce(IMonoid<T> monoid) {
        var result = monoid.Zero();
        for (var i = 0; i < Values.Length; i++)
        {
            result = monoid.Op(result, Values[i]);
        }
        return result;
    }
}

File: monoid-folding-test-typescript/src/MonoidFoldingTest/Series.ts
import { IMonoid } from "./IMonoid";

export class Series<T> {
    constructor(private values: T[]) {
    }

    reduce(monoid: IMonoid<T>): T {
        let result: T = monoid.zero();
        for (let i = 0; i < this.values.length; i++) {
            result = monoid.op(result, this.values[i]);
        }
        return result;
    }
}
